Assets/Tests/EditModeTests/Server/HandleDisconnectSystemTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿using NUnit.Framework;
+using PropHunt.Mixed.Components;
+using PropHunt.Server.Systems;
+using Unity.Entities;
+using Unity.Entities.Tests;
+using Unity.NetCode;
+
+namespace PropHunt.EditMode.Tests.Server
+{
+    /// <summary>
+    /// Tests to verify functionality of the ConnectionSystem
+    /// </summary>
+    [TestFixture]
+    public class HandleDisconnectSystemTests : ECSTestsFixture
+    {
+        /// <summary>
+        /// Connection system tests
+        /// </summary>
+        private HandleDisconnectSystem handleDisconnectSystem;
+
+        [SetUp]
+        public override void Setup()
+        {
+            base.Setup();
+            this.handleDisconnectSystem = base.World.CreateSystem<HandleDisconnectSystem>();
+        }
+
+        /// <summary>
+        /// Craete a disconnect for a given Id
+        /// </summary>
+        /// <param name="id">Id of player to disconnect</param>
+        /// <returns>Entity that contains the disconnect event</returns>
+        public Entity CreateDisconnectEvent(int id)
+        {
+            Entity disconnectEvent = base.m_Manager.CreateEntity();
+            base.m_Manager.AddComponent<NetworkStreamDisconnected>(disconnectEvent);
+            base.m_Manager.AddComponentData<NetworkIdComponent>(disconnectEvent, new NetworkIdComponent { Value = id });
+            return disconnectEvent;
+        }
+
+
+        /// <summary>
+        /// Craete a player with a given Id
+        /// </summary>
+        /// <param name="id">Id to assign to player</param>
+        /// <returns>Entity that contains the player</returns>
+        public Entity CreatePlayer(int id)
+        {
+            Entity player = base.m_Manager.CreateEntity();
+            base.m_Manager.AddComponentData<PlayerId>(player, new PlayerId { playerId = id });
+            return player;
+        }
+
+        /// <summary>
+        /// Verify behaviour of handle disconnect system when there is one playerID object and 
+        /// no disconnect event
+        /// </summary>
+        [Test]
+        public void OnePlayerNoDisconnect()
+        {
+            // Create a player with id = 0
+            Entity player0 = this.CreatePlayer(0);
+
+            // Update the handle disconnect system
+            this.handleDisconnectSystem.Update();
+
+            // Verify player still exists
+            Assert.IsTrue(base.m_Manager.Exists(player0));
+        }
+
+        /// <summary>
+        /// Verify behaviour of handle disconnect system when there is one playerID object and 
+        /// one disconnect event
+        /// </summary>
+        [Test]
+        public void OnePlayerOneDisconnect()
+        {
+            // Create a disconnect event with id = 0
+            Entity disconnect = this.CreateDisconnectEvent(0);
+            // Create a player with id = 0
+            Entity player0 = this.CreatePlayer(0);
```

Comment:
This method (createPlayer) could be useful for other unit tests I guess.
Also it'll be nice to centralize the logic of it, if in the future the structure of the player changes, it'll be nice to have a central way of dispatching this data, instead of segregated and specialized troughout the files.

Reply:
On a second thought, I think for unit testing it's better if we specialize the structures to just the data that we need.
Ignore my previous comment.